src/array_api_stubs/_draft/searching_functions.py

```diff
@@ -168,6 +176,50 @@ def searchsorted(
     """
 
 
+def top_k(
```

Comment:
Generally this look good. Other implementations I've seen pass a DeviceContext as well but I'm not sure if we want to tackle that as part of the initial implementation.